public/scripts/generators/data_generator.js

```diff
@@ -0,0 +1,208 @@
+
+    {distribution_type: "gaussian", round_type: 'test', trials_per_round: 4, high_ref: 0.5, low_ref: 0.25, error: 0.0001, num_points: 100, regen_rate: 1000, mean: 0.5, SD: 0.2, num_SD: 2.5, point_color: 'BLACK', axis_color: 'BLACK', text_color: 'BLACK', background_color: 'WHITE', point_size: 3, regen_rate: 1000},
+
+    {distribution_type: "gaussian", round_type: 'test', trials_per_round: 4, high_ref: 0.75, low_ref: 0.5, error: 0.0001, num_points: 100, regen_rate: 1000, mean: 0.5, SD: 0.2, num_SD: 2.5, point_color: 'BLACK', axis_color: 'BLACK', text_color: 'BLACK', background_color: 'WHITE', point_size: 3, regen_rate: 1000},
+
+    {distribution_type: "gaussian", round_type: 'test', trials_per_round: 4, high_ref: 1, low_ref: 0.75, error: 0.0001, num_points: 100, regen_rate: 1000, mean: 0.5, SD: 0.2, num_SD: 2.5, point_color: 'BLACK', axis_color: 'BLACK', text_color: 'BLACK', background_color: 'WHITE', point_size: 3, regen_rate: 1000}
+    ],
+
+  design: []
+}
+
+const STEVENS_CONDITIONS = {
+
+}
+
+/**
+ * Retrieves the data given experiment, range, and condition.
+ * Creates a shortened version if is_full_set = false.
+ *
+ * @param  experiment  {string}            "jnd" or "stevens"   
+ *         range       {string}            "foundational" or "design"         
+ *         condition   {string}            Name of condition
+ *         is_full_set {boolean}           If true, will use entire data set
+ *
+ * @return dataset     [{assoc}, {assoc}, .... ]         
+ */
+function get_data(experiment, range, condition, is_full_set){
+  var dataset;
+
+  if (experiment === "jnd"){
+
+    dataset = JND_BASE[range];
+
+    if (condition !== "base") {
+      dataset = append_condition_data(dataset, JND_CONDITIONS, condition);
+    }
+  }
+  else if (experiment === "stevens"){
+
+    dataset = STEVENS_BASE[range];
+
+    if (condition !== "base") {
+      dataset = append_condition_data(dataset, STEVENS_CONDITIONS, condition);
+    }
+  }
+  else {
+    throw new Error(experiment + " not supported.");
+  }
+
+  // Splice out only 4 subconditions if don't want full set
+  if (!is_full_set){
+    dataset = dataset.slice(0,4);
+  }
+
+  return dataset;
+}
+
+/**
+ * Appends condition-specific data to the dataset.
+ *
+ * @param  dataset           [{assoc}, {assoc}, .... ]               dataset with base experiment constants   
+ *         condition_data    {condition: [], condition2: []... }     condition set for that experiment         
+ *         condition         {string}                                Name of condition
+ *
+ * @return dataset           [{assoc}, {assoc}, .... ]  
+ **/
+function append_condition_data(dataset, condition_data, condition){
```

Comment:
why do you need `condition_data` and `condition`? it seems like you can just use condition_data[condition]